pkg/transport/process/pipeline.go

```diff
@@ -0,0 +1,83 @@
+package process
+
+import (
+	"context"
+	"os"
+
+	"archive/tar"
+	"fmt"
+	"io/ioutil"
+
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+)
+
+type resourceProcessingPipelineImpl struct {
+	processors []ResourceStreamProcessor
+}
+
+func (p *resourceProcessingPipelineImpl) Process(ctx context.Context, cd cdv2.ComponentDescriptor, res cdv2.Resource) (*cdv2.ComponentDescriptor, cdv2.Resource, error) {
+	infile, err := ioutil.TempFile("", "out")
```

Comment:
we might want to use the virtual filesystem lib to better test it?

Reply:
why do you think we need the vfs lib here for testing? I just tested this component as a black box with 2 test processors